### PR TITLE
GAE standard bookshelf - step 4

### PR DESCRIPTION
**Pull request by jerjou**

### CODE REVIEW

File: bookshelf-standard/4-auth/src/main/java/com/example/getstarted/auth/Oauth2CallbackServlet.java [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    req.getSession().removeAttribute("state");     // Remove one-time use state.
+
+    flow = new GoogleAuthorizationCodeFlow.Builder(
+        HTTP_TRANSPORT,
+        JSON_FACTORY,
+        getServletContext().getInitParameter("bookshelf.clientID"),
+        getServletContext().getInitParameter("bookshelf.clientSecret"),
+        SCOPES).build();
+
+    final TokenResponse tokenResponse =
```
Comment (lesv) [inline bookshelf-standard/4-auth/src/main/java/com/example/getstarted/auth/Oauth2CallbackServlet.java]: Shouldn't all this stuff use the GAE Standard USER API??

PTAL

**Comment by lesv** [comment]
Also, have you tried deploying?

File: bookshelf-standard/4-auth/src/main/java/com/example/getstarted/auth/Oauth2CallbackServlet.java (hunk 1 above)
Comment (jerjou) [inline bookshelf-standard/4-auth/src/main/java/com/example/getstarted/auth/Oauth2CallbackServlet.java]: Oh yeah - I wasn't sure if showing the oauth2 flow was part of the point of this tutorial, so I left it in. Agree that it'd be better with the user api. Will update.

**Comment by jerjou** [comment]
Updated to use Users api. PTAL

**Pull request merged**